core_types/src/dna/traits.rs

```diff
@@ -0,0 +1,58 @@
+//! Module for zome trait structures
+
+use std::str::FromStr;
+
+#[derive(Debug, PartialEq)]
+/// Enumeration of all Traits known and used by HC Core
+/// Enumeration converts to str
+pub enum ReservedTraitNames {
+    /// Development placeholder, no production fn should use MissingNo
+    MissingNo,
+
+    /// used for declaring functions that will auto-generate a public grant during genesis
+    Public,
+}
+
+impl FromStr for ReservedTraitNames {
+    type Err = &'static str;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "hc_public" => Ok(ReservedTraitNames::Public),
```

Comment:
can you define "hc_public" as a const that you reference everywhere you currently use the magic string?

Reply:
Adding it that way instead also serves as a reminder to document the nature of that string, and what it represents, whereas the use of the raw string throughout the code doesn't

Reply:
The problem with this is there is an issue with how this works in macros.  In the macros the name can't have quotes.  I.e. the macro adds quotes:
```
    traits: {
        hc_public [call_bridge]
    }
```
So you couldn't use any static like HC_PUBLIC because that would turn into "HC_PUBLIC"   We would have to change the way the macro works.  I could put that in for internal use, but then it would be a half measure.. so I would prefer to leave it "hc_public" so you can search and find them the same way.

Reply:
K. happy to let this one go, just holding for this to happen more going forward